ProjectHive/ProjectHive.Services.AuthAPI/Controllers/UserController.cs

```diff
@@ -33,11 +35,17 @@ public async Task<IActionResult> DeleteById(Guid id, CancellationToken cancellat
     [HttpPost("[action]")]
     public async Task<IActionResult> CreateUser(RegisterModel request, CancellationToken cancellationToken)
     {
-        var dto = mapper.Map<UserDto>(request);
-
-        await userService.RegisterUser(dto, cancellationToken);
-        var user = await userService.GetByEmail(request.Email, cancellationToken);
-        return Created($"users/{user.Id}", user);
+        if ((await userRegisterValidator.ValidateAsync(request)).IsValid)
+        {
```

Comment:
Tsi is ok but also fluent validator allows to use it implicitly

https://dev.to/andytechdev/how-to-use-fluentvalidation-in-aspnet-core-net-6-2fnf